zummon/src/commonMain/kotlin/companysurvey/Validation.kt

```diff
@@ -65,6 +70,38 @@ class GridConnectionValidator : Validator<GridConnection> {
             listOf(ValidationResult(Status.INVALID, translate("gridConnection.totalPowerChargePointsInvalid", totalPowerChargePoints, contractedCapacity + batteryPower)))
         }
     }
+
+    //quarter-hourly production should not be all zeroes if hasPv is true
+    fun validateQuarterHourlyFeedIn(gridConnection: GridConnection): ValidationResult {
+        return when {
+            gridConnection.supply.hasSupply == true && (gridConnection.electricity.quarterHourlyProduction_kWh?.values?.all { it == 0f } == true) ->
+                ValidationResult(Status.INVALID, translate("electricity.quarterHourlyProductionCannotBeAllZero"))
+            else ->
+                ValidationResult(Status.VALID, translate("electricity.quarterHourlyProductionValid"))
+        }
+    }
+
+    //every time step in quarter-hourly feed-in should be less than or equal to production + battery power
```

Comment:
I think we mean to compare every data point in the feed-in time series to every point in the the production time series.